intermine/query_manager.py

```diff
@@ -0,0 +1,198 @@
+    Enter the mine name: <enter mine name>
+"""
+
+def get_all_query_names():
+    """
+    A function to list all the queries that are saved in a user account
+    ================================================
+    example:
+
+        >>>from intermine import query_manager as qm
+        >>>qm.get_all_query_names()
+        <returns the names of all the saved queries in user account>
+
+    """
+    x = "http://registry.intermine.org/service/instances/" + mine
+    r = requests.get(x)
+    dict = yaml.load(r.text)
+    link = dict["instance"]["url"] + "/service/user/queries?token=" + token
+    r = requests.get(link)
+    dict = yaml.load(r.text)
+    count = 0
+    for key in dict['queries'].keys():
+        count = count + 1
+        print(key)
+
+    if(count == 0):
+        print("No saved queries")
+
+def get_query(name):
+    """
+    A function that returns the columns that a given query constitutes of
+    ================================================
+    example:
+
+        >>>from intermine import query_manager as qm
+        >>>qm.get_query('queryName')
+        <returns information about the query whose name is 'queryName'>
+
+    """
+    x = "http://registry.intermine.org/service/instances/" + mine
+
+    r = requests.get(x)
+    dict = yaml.load(r.text)
+    link = dict["instance"]["url"] + "/service/user/queries?token=" + token
+
+    r = requests.get(link)
+    dict = yaml.load(r.text)
+    count = 0
+    for key in dict['queries'].keys():
+        if(name == key):
+            count = count + 1
+            print("Columns:")
+            for i in range(len(dict['queries'][name]['select'])):
+                print(dict['queries'][name]['select'][i])
+    if(count == 0):
+        print("No such query available")
+
+
+def delete_query(name):
+    """
+    A function that deletes a given query
+    ================================================
+    example:
+
+        >>>from intermine import query_manager as qm
+        >>>qm.delete_query('queryName')
+        <deletes the query whose name is 'queryName' from user's account>
+
+    """
+    x = "http://registry.intermine.org/service/instances/" + mine
+    r = requests.get(x)
+    dict = yaml.load(r.text)
+
+    y = dict["instance"]["url"] + "/service/user/queries?token=" + token
+    r = requests.get(y)
+    z = yaml.load(r.text)
+    count = 0
+    for key in z['queries'].keys():
+        if (key == name):
+            count = count + 1
+    if(count == 0):
+        print("No such query available")
+
+    else:
+        link = dict["instance"]["url"] + "/service/user/queries/" + name +\
+         "?token=" + token
+        requests.delete(link)
+
+    '''
+    x = "http://registry.intermine.org/service/instances/" + mine
+    r = requests.get(x)
+    dict = yaml.load(r.text)
+    link = dict["instance"]["url"] + "/service/user/queries/" + name +\
+     "?token=" + token
+    requests.delete(link)
+    '''
+
+def xml_to_link(xml):
```

Comment:
To my non-python eyes, this looks fragile to me. Should we use a library instead?

https://stackoverflow.com/questions/5607551/how-to-urlencode-a-querystring-in-python

Reply:
Any of these libraries is not fully satisfying our requirements. For instance 'urllib.request' isn't converting "/" to "%2F". 
Also, I was avoiding other dependencies( although urllib.request is inbuilt in python3 ) 
So, even if we find such a library it would just make things complicated.

Reply:
@justinccdev What are your thoughts? I would prefer to not have a lot of dependencies, but XML is pants.

Reply:
Well, 'urllib.request' is doing the job right. (inspite of the fact that it doesn't convert "/" ).
So, its your choice whether we should use this inbuilt library or define a function in the code itself.